### ISSUE TITLE
move to scikit-learn-contrib?

### ISSUE THREAD

**Issue by amueller** [satra/mapalign#7]
I think this would be a great project to move to scikit-learn-contrib and maybe from there to scikit-learn at some point:
https://github.com/scikit-learn-contrib/scikit-learn-contrib/blob/master/README.md

**Comment by satra** [comment]
@amueller - this is such a quiet repo i did not see this at all amongst my many github emails!

i think it would be good to move it scikit-learn-contrib.  we also need to make this fully scikit-learn compatible.

**Comment by amueller** [comment]
I know what that's like ;)

**Comment by jolespin** [comment]
Any updates on getting this integrated into scikitlearn? @amueller

**Comment by satra** [comment]
@jolespin - if you are interested in sending a PR that would be great. there was a PR a long time back and it was decided that it did not fit the criteria in some ways. i don't think that's necessarily the case with diffusion maps anymore. this repo simply became a placeholder for a method that we used and needed.

**Comment by satra** [comment]
just as a quick addition, there are two components to this repo.

1. diffusion map algo
2. alignment methods

part 2 is still a fairly open space of research of how different embeddings could be mapped on to each other. and this repo only implements a procrustes alignment, optionally iterative. there are other projects that implement others.

**Comment by jolespin** [comment]
@satra  Do you have any experience with the pydiffmap package? I'm looking into using diffusion maps for my genomics research and trying to figure out which implementation is the most appropriate. Your method seemed to be the closest thing to an official Scikit diffusion map release.

Also does this approach implement classical or pseudo time diffusion maps?

**Comment by satra** [comment]
regarding pydiffmap, i took a look and it looks like an interesting implementation. it does so without doing an eigen decomposition directly. may want to ask there if it's still being maintained.

thanks for the pointer to the landmark diffusion paper on the scikit-learn thread. 

nice, there is another approximation here that could be applied to diffusion maps as well: https://www.nature.com/articles/s41592-023-02139-9

one of the original intents of this was to be able to expand out the map-align transform. given all the work going on in single-cell genomics this may be worthwhile still to bring together a domain agnostic methods package that can leverage and compare different approaches.

if you are interested i'm happy to add you as a collaborator to this repo. i'll also move it to our group's org. let me know.

**Comment by jolespin** [comment]
I've been working on adding approximate k-nearest neighbor searches to speed up mapalign. The next step on the list is to add out of sample transformation which is essential for my research.  If I'm successful I would be more than happy to integrate it into the main repository. 

Quick followup, I found this package last night: https://dbmap.readthedocs.io/en/latest/docs/#dbmap.diffusion

> dbMAP implements the Diffusor() class, which allows state-of-the-art dimensional reduction by the fast approximation of the Laplace Beltrami operator and automatic detection of intrinsic dimensionality. This algorithm learns a local metric which is normalized and embedded as a diffusion distance on the series of orthogonal components that define structure variability within the initial informational space. 
> ...
> Sklearn estimator for using fast anisotropic diffusion with a multiscaling adaptive algorithm as proposed by Setty et al, 2018, and optimized by Sidarta-Oliveira, 2020.

@satra **Does mapalign implement anisotropic diffusion?**

**Comment by satra** [comment]
> Does mapalign implement anisotropic diffusion?

it doesn't. just vanilla diffusion.
